### PR TITLE
Remove reconnect from C++ Client

### PR DESCRIPTION
**Pull request by BrennanConroy**
Starting to cleanup the client.
Removed all the reconnect stuff because it was a big chunk of code that isn't used and the reconnect tests were causing test crashes.
Cleaned up a bunch of tests to expect the new protocol instead of the old one. We were at ~100 failing tests before now down to ~25.

### CODE REVIEW

File: src/SignalR/clients/cpp/src/signalrclient/connection_impl.cpp [hunk 1]
```diff
@@ -40,7 +40,7 @@ namespace signalr
         : m_base_url(url), m_query_string(query_string), m_connection_state(connection_state::disconnected), m_reconnect_delay(2000),
         m_logger(log_writer, trace_level), m_transport(nullptr), m_web_request_factory(std::move(web_request_factory)),
         m_transport_factory(std::move(transport_factory)), m_message_received([](const web::json::value&){}),
-        m_reconnecting([](){}), m_reconnected([](){}), m_disconnected([](){}), m_handshakeReceived(false)
+        m_disconnected([](){}), m_handshakeReceived(false)
     { }
 
     connection_impl::~connection_impl()
```
Comment (mikaelm12) [inline src/SignalR/clients/cpp/src/signalrclient/connection_impl.cpp]: The large comment block below has a few references to reconnect.

File: src/SignalR/clients/cpp/src/signalrclient/connection_impl.cpp [hunk 2]
```diff
@@ -40,7 +40,7 @@ namespace signalr
         : m_base_url(url), m_query_string(query_string), m_connection_state(connection_state::disconnected), m_reconnect_delay(2000),
```
Comment (mikaelm12) [inline src/SignalR/clients/cpp/src/signalrclient/connection_impl.cpp]: `m_reconnect_delay(2000)` here

**Review by mikaelm12 — COMMENTED** [review]
There are also other comments that have references to reconnect. Do a global search for the term.

File: src/SignalR/clients/cpp/src/signalrclient/connection_impl.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (mikaelm12) [reply src/SignalR/clients/cpp/src/signalrclient/connection_impl.cpp]: And then follow where this flows

File: src/SignalR/clients/cpp/src/signalrclient/hub_connection_impl.cpp [hunk 3]
```diff
@@ -270,27 +244,11 @@ namespace signalr
...
                         std::make_exception_ptr(
-                            signalr_exception(message.at(_XPLATSTR("error")).serialize())));
+                            hub_exception(message.at(_XPLATSTR("error")).serialize())));
                 }
 
                 set_result(json::value::null());
             };
         }
-
-        static utility::string_t adapt_url(const utility::string_t& url, bool use_default_url)
```
Comment (halter73) [inline src/SignalR/clients/cpp/src/signalrclient/hub_connection_impl.cpp]: Why was this needed before but not now?

Comment (BrennanConroy) [reply src/SignalR/clients/cpp/src/signalrclient/hub_connection_impl.cpp]: I think I removed it's usage a long time ago when I ported the client. It's not referenced anywhere which is why I removed it here.

**Comment by BrennanConroy** [comment]
Ping

File: src/SignalR/clients/cpp/src/signalrclient/url_builder.cpp [hunk 4]
```diff
@@ -80,7 +80,7 @@ namespace signalr
             const utility::string_t& connection_data, const utility::string_t& query_string,
             const utility::string_t& last_message_id = _XPLATSTR(""), const utility::string_t& groups_token = _XPLATSTR(""))
         {
-            _ASSERTE(command == _XPLATSTR("reconnect") || (last_message_id.length() == 0 && groups_token.length() == 0));
```
Comment (halter73) [inline src/SignalR/clients/cpp/src/signalrclient/url_builder.cpp]: There are still a bunch of commented-out reconnect tests in url_builder_tests.cpp. Can we remove the entire url_builder_tests.cpp file? Or at least the reconnect test portions of it?

File: src/SignalR/clients/cpp/test/signalrclient-e2e-tests/hub_connection_tests.cpp [hunk 5]
```diff
@@ -331,17 +236,6 @@ TEST(hub_connection_tests, connection_id_start_stop_start_reconnect)
     ASSERT_NE(hub_conn->get_connection_id(), connection_id);
 
     connection_id = hub_conn->get_connection_id();
-
-    try
-    {
-        hub_conn->send(U("forceReconnect")).get();
```
Comment (halter73) [inline src/SignalR/clients/cpp/test/signalrclient-e2e-tests/hub_connection_tests.cpp]: It looks like nothing calls HubConnection.ForceReconnect anymore in signalrclient-testhost. Should it be removed?

**Review by halter73 — CHANGES_REQUESTED** [review]

**Review by halter73 — APPROVED** [review]
We can take care of the nits in #7160
